src/components/Shared/SendModal/SendOnChain/SendOnChain.tsx

```diff
@@ -1,82 +1,117 @@
-import { ChangeEvent, FC, FormEvent, useContext } from "react";
+import { FC, useContext } from "react";
+
+import type { ChangeEvent } from "react";
+
 import { useTranslation } from "react-i18next";
 import { AppContext } from "../../../../store/app-context";
 import AmountInput from "../../AmountInput/AmountInput";
+import InputField from "../../InputField/InputField";
+
+import { useForm } from "react-hook-form";
+import type { SubmitHandler } from "react-hook-form";
+
+interface IFormInputs {
+  addressInput: string;
+  feeInput: number;
+  amountInput: number;
+  commentInput: string;
+}
 
 const SendOnChain: FC<SendOnChainProps> = (props) => {
   const { t } = useTranslation();
   const appCtx = useContext(AppContext);
   const {
-    balance,
     address,
-    onChangeAddress,
     amount,
-    onChangeAmount,
-    fee,
-    onChangeFee,
+    balance,
     comment,
+    fee,
+    onChangeAddress,
+    onChangeAmount,
     onChangeComment,
+    onChangeFee,
     onConfirm,
   } = props;
 
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, submitCount },
+  } = useForm<IFormInputs>({
+    mode: "onChange",
+  });
+
+  const onSubmit: SubmitHandler<IFormInputs> = (_data) => onConfirm();
+
   return (
-    <form className="px-5" onSubmit={onConfirm}>
+    <form className="px-5" onSubmit={handleSubmit(onSubmit)}>
       <h3 className="text-xl font-bold">{t("wallet.send_onchain")}</h3>
-      <div className="my-5">
+
+      <p className="my-5">
         <span className="font-bold">{t("wallet.balance")}:&nbsp;</span>
         {balance} {appCtx.unit}
-      </div>
-      <div className="my-5 flex flex-col justify-center text-center items-center">
+      </p>
+
+      <fieldset className="my-5 flex flex-col justify-center text-center items-center">
         <div className="w-full md:w-10/12 py-1">
-          <label htmlFor="address" className="label-underline">
-            {t("wallet.address")}
-          </label>
-          <input
-            id="address"
-            type="text"
-            className="w-full input-underline"
+          <InputField
+            {...register("addressInput", {
+              required: t("forms.validation.chainAddress.required") as string,
+              pattern: {
+                value: /^(1|3|bc1)\w+/i,
```

Comment:
This should allow testnet addresses (tb1& tpub) and regtest addresses (bcrt...) as well

See also https://en.bitcoin.it/wiki/List_of_address_prefixes